engine/uState.ml

```diff
@@ -1030,11 +1030,21 @@ let demote_seff_univs univs uctx =
 
 let demote_global_univs env uctx =
   let env_ugraph = Environ.universes env in
-  let global_univs = UGraph.domain env_ugraph in
-  let global_constraints, _ = UGraph.constraints_of_universes env_ugraph in
-  let promoted_uctx =
-    ContextSet.(of_set global_univs |> add_constraints global_constraints) in
-  { uctx with local = ContextSet.diff uctx.local promoted_uctx }
+  let mem_univ u ugraph = match UGraph.check_declared_universes ugraph (Level.Set.singleton u) with
+  | () -> true
+  | exception (UGraph.UndeclaredLevel _) -> false
```

Comment:
should we add some `mem_level` API to ugraph instead of messing with sets and exceptions here?

Reply:
It's a pretty low-level function, I'm not sure we want to expose more hacking primitives to the user.